### PR TITLE
responsive-styles

### LINKED ISSUE #8: Add styles for mobile and tablet screens
Requirements:

- For desktop view - render a table
- For mobile view - render a list. 
    - For the current weather forecast render it should be nested - at first level render a date, at the second - 3-hour forecase. 
    - For the historical data - a list item shoul consist of date in the left part and weather data in the right.

### PR DESCRIPTION
**Pull request by coding-vibe**

### CODE REVIEW

File: src/components/Forecast/Forecast.tsx [hunk 1]
```diff
@@ -29,9 +31,14 @@ export default function Forecast() {
   } = useContext<SettingsContextType>(SettingsContext);
   const [selectedLocation, onSelectLocation] = useState<Location | null>(null);
   return (
-    <>
+    <Box css={classes.mainBox}>
+      <Box css={classes.box}>
```
Comment (singvarr) [inline src/components/Forecast/Forecast.tsx:35]: You don't need it. Just put styles to `p`

File: src/components/Forecast/Forecast.tsx [hunk 2]
```diff
@@ -29,9 +31,14 @@ export default function Forecast() {
...
+        <p css={classes.text}>
```
Comment (singvarr) [inline src/components/Forecast/Forecast.tsx]: Use MUI `Typography`

File: src/components/Forecast/styles.ts [hunk 3]
```diff
@@ -1,13 +1,20 @@
 import { css } from '@emotion/react';
 
-export const languageSelect = css`
-  margin-top: 10px;
+export const mainBox = css`
+  display: flex;
+  flex-direction: column;
+  gap: 30px;
+
+  margin-top: 30px;
```
Comment (singvarr) [inline src/components/Forecast/styles.ts]: It's better to put padding and they should be symmetric

File: src/components/Forecast/styles.ts [hunk 4]
```diff
@@ -1,13 +1,20 @@
 import { css } from '@emotion/react';
 
-export const languageSelect = css`
-  margin-top: 10px;
+export const mainBox = css`
+  display: flex;
+  flex-direction: column;
+  gap: 30px;
```
Comment (singvarr) [inline src/components/Forecast/styles.ts]: You don't need this flex. Add margins for elements of container individually

File: src/utils/formatDate.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { format } from 'date-fns';
+
+const convertTimestampToDate = (unixDate: number): Date => {
+  const date = new Date(unixDate * 1000);
+  return date;
+};
+export default convertTimestampToDate;
+
+export const formatDate = (date: Date): string => {
+  const formattedDate = format(date, 'dd MMMM');
```
Comment (singvarr) [inline src/utils/formatDate.ts]: to const

File: src/types/forecast.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  ];
+}
+
+export default ForecastBody;
+
+export interface ForecastAPIResponse {
+  list: ForecastBody[];
+}
+
+export type Forecast = Array<[string, ForecastBody[]]>;
```
Comment (singvarr) [inline src/types/forecast.ts]: `[string, ForecastBody[]][]`

File: src/components/WeatherWidget/styles.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { css } from '@emotion/react';
+
+export const weatherTable = css`
+  @media (max-width: 999px) {
+    display: none;
+  }
+`;
+
+export const weatherList = css`
+  @media (min-width: 1000px) {
```
Comment (singvarr) [inline src/components/WeatherWidget/styles.ts]: use MUI `theme.breakpoints.down('lg')`

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 8]
```diff
@@ -86,76 +64,24 @@ export default function WeatherWidget({ location }: Props) {
...
-                  </td>
-                );
-              })}
-            </tr>
-          );
-        })}
-      </tbody>
-    </table>
+    <Box>
+      {forecast && (
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: 1. Don't duplicate this check. Also convert `forecast` to boolean explicitly
2. Don't render `Box` if there is no content

File: src/components/WeatherTable/WeatherTable.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+import HOURS from './hours';
+import * as classes from './styles';
+
+interface Props {
+  forecast: Forecast;
+  location: Location;
+  className?: string;
+}
+
+const TIME_PERIODS_AMOUNT = 8;
```
Comment (singvarr) [inline src/components/WeatherTable/WeatherTable.tsx]: Pick up better name. Also can be derived from `HOURS.length`

File: src/components/WeatherTable/WeatherTable.tsx [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+                  css={classes.tableHeadCell}
+                  key={hour}>
+                  {hour}
+                </MUITableCell>
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {forecast?.map(([date, weather], index) => {
+              const emptyCells = TIME_PERIODS_AMOUNT - weather.length;
```
Comment (singvarr) [inline src/components/WeatherTable/WeatherTable.tsx]: What `weather.length` mean?

File: src/components/WeatherList/styles.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
+import { css } from '@emotion/react';
+
+export const headList = css`
+  border: 1px solid #f5f5f5;
```
Comment (singvarr) [inline src/components/WeatherList/styles.ts]: remove blank lines between CSS rules (important - not nested classes)

File: src/components/WeatherList/styles.ts [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+export const mainListItemText = css`
+  & .MuiListItemText-primary {
+    color: #49484a;
+
+    font-size: 16px;
+    font-weight: 800;
+  }
+`;
+export const listItem = css`
```
Comment (singvarr) [inline src/components/WeatherList/styles.ts]: add blank line between exports

File: src/components/WeatherList/styles.ts [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+export const headList = css`
+  border: 1px solid #f5f5f5;
+
+  width: 420px;
+
+  & .MuiListSubheader-root {
+    border-bottom: 1px solid #f5f5f5;
+
+    color: #49484a;
```
Comment (singvarr) [inline src/components/WeatherList/styles.ts]: If it's reused, it should be in theme

File: src/components/WeatherList/WeatherList.tsx [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+interface Props {
+  forecast: Forecast;
+  location: Location;
+  className?: string;
+}
+
+export default function WeatherList({ forecast, location, className }: Props) {
+  const { selectedTemperatureUnit } =
+    useContext<SettingsContextType>(SettingsContext);
+  const [openItems, setOpenItems] = useState<boolean[]>(
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx]: Do you need this state at all?

File: src/components/WeatherList/WeatherList.tsx [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+        <ListSubheader component='div'>
+          5-Day Weather Forecast for&nbsp;
+          {location &&
+            `${findCountryNameByCode(location.country)}, ${location.name}`}
+        </ListSubheader>
+      }>
+      {forecast?.map(([date, weather], index) => (
+        <List
+          key={date}
+          component='nav'
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx]: It's not nav

File: src/components/WeatherList/WeatherList.tsx [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+            {weather.map((hourlyWeather, idx) => {
+              const {
+                dt,
+                main: { humidity, temp },
+                weather: [{ description, icon }],
+              } = hourlyWeather;
+              const hour = format(dt * 1000, 'hh:00 a');
+              return (
+                <List
+                  component='div'
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx]: should be `ul`

File: src/components/TableCell/TableCell.tsx [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <MUITableCell
+      css={classes.tableCell}
+      key={dt}>
+      {displayDate && formatDate(convertTimestampToDate(dt))}
+      <Tooltip title={description}>
+        <img
+          alt='Weather condition'
+          css={classes.icon}
+          src={`${import.meta.env.VITE_BASE_URL}img/wn/${icon}.png`}
```
Comment (singvarr) [inline src/components/TableCell/TableCell.tsx]: Move function that constructs icon URL to utils. It's reused

File: src/components/TableCell/TableCell.tsx [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+        />
+      </Tooltip>
+      <p>
```
Comment (singvarr) [inline src/components/TableCell/TableCell.tsx]: Just span

File: src/components/TableCell/TableCell.tsx [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+import convertTimestampToDate, { formatDate } from 'utils/formatDate';
+import formatTemperatureData from 'utils/formatTemperature';
+import * as classes from './styles';
+
+interface TableCellProps {
+  weather: ForecastBody;
+  displayDate?: boolean;
+}
+
+export default function TableCell({ weather, displayDate }: TableCellProps) {
```
Comment (singvarr) [inline src/components/TableCell/TableCell.tsx]: `displayDate` -> `isDateShown`

File: src/components/TableCell/styles.ts [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+export const tableCell = css`
+  font-size: 13px;
+`;
+
+export const icon = css`
+  margin: 0 auto;
+`;
+
+export const data = css`
+  color: #49484a;
```
Comment (singvarr) [inline src/components/TableCell/styles.ts]: Put me in theme and take me from there

File: src/components/TableCell/TableCell.tsx [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+          src={`${import.meta.env.VITE_BASE_URL}img/wn/${icon}.png`}
+        />
+      </Tooltip>
+      <p>
+        Temp:&nbsp;
+        <span css={classes.data}>
+          {formatTemperatureData(temp, selectedTemperatureUnit)}
+        </span>
+      </p>
+      <p>
```
Comment (singvarr) [inline src/components/TableCell/TableCell.tsx]: Can we use MUI typography?

File: src/components/Select/styles.ts [hunk 22]
```diff
@@ -1,5 +1,5 @@
 import { css } from '@emotion/react';
 
 export const select = css`
-  min-width: 200px;
+  width: 420px;
```
Comment (singvarr) [inline src/components/Select/styles.ts]: Will it work on smaller devices?

File: src/components/Select/Select.tsx [hunk 23]
```diff
@@ -24,15 +23,12 @@ export default function Select<Option extends OptionBase>({
   options,
   setValue,
   value,
-  className,
```
Comment (singvarr) [inline src/components/Select/Select.tsx]: It was a good change. Keep it even if you don't pass any className now

File: src/components/LocationAutocomplete/LocationAutocomplete.tsx [hunk 24]
```diff
@@ -146,7 +143,6 @@ export default function LocationAutocomplete({
 }
 
 LocationAutocomplete.defaultProps = {
-  className: null,
```
Comment (singvarr) [inline src/components/LocationAutocomplete/LocationAutocomplete.tsx]: Also revert this change

**Review by singvarr — COMMENTED** [review]
Quite a lot of comments. But they are merely about codestyle

**Comment by singvarr** [comment]
@coding-vibe, add more descriptive title to PR

**Comment by singvarr** [comment]
Make `emotion` theme compatible with MUI and remove `any` in your styles. In order to achieve this extend `Theme` from the first package by interface from MUI like this:
```ts
declare module '@emotion/react' {
  interface Theme extends MuiTheme {}
}
```

File: src/components/ThemeProvider/ThemeProvider.tsx [hunk 25]
```diff
@@ -0,0 +1,54 @@
...
+} from '@mui/material/styles';
+
+interface Props {
+  children: ReactNode;
+}
+
+const ThemeProvider: FC<Props> = function ThemeProvider({ children }) {
+  const theme = createTheme({
+    breakpoints: {
+      values: {
```
Comment (singvarr) [inline src/components/ThemeProvider/ThemeProvider.tsx]: Do yo need these overrides of breakpoints and typography?

File: src/components/ThemeProvider/ThemeProvider.tsx [hunk 26]
```diff
@@ -0,0 +1,54 @@
...
+  createTheme,
+  ThemeProvider as MuiThemeProvider,
+} from '@mui/material/styles';
+
+interface Props {
+  children: ReactNode;
+}
+
+const ThemeProvider: FC<Props> = function ThemeProvider({ children }) {
+  const theme = createTheme({
```
Comment (singvarr) [inline src/components/ThemeProvider/ThemeProvider.tsx]: Move me out of component. If it's inside and `ThemeProvider` is rerendered performance of the app may degrade significantly
After this change you don't need this component at all. Just import provider and theme in entry point

File: src/components/ThemeProvider/ThemeProvider.tsx [hunk 27]
```diff
@@ -0,0 +1,54 @@
...
+      overline: {
+        fontSize: '13px',
+        fontWeight: 500,
+      },
+      subtitle1: {
+        fontSize: '20px',
+        fontWeight: 800,
+      },
+      subtitle2: {
+        fontSize: '13px',
```
Comment (singvarr) [inline src/components/ThemeProvider/ThemeProvider.tsx]: You can not specify `px`. Just leave numbers

File: src/components/TableCell/TableCell.tsx [hunk 28]
```diff
@@ -22,40 +25,44 @@ export default function TableCell({ weather, displayDate }: TableCellProps) {
...
-  };
-  const formatTemperatureData = (
-    temperature: number,
-    temperatureUnit: TemperatureUnits,
-  ) => `${Math.floor(temperature)}${formatTemperatureUnits(temperatureUnit)}`;
-
   return (
-    <td key={dt}>
-      {displayDate && formatDate(convertTimestampToDate(dt))}
+    <MUITableCell key={dt}>
```
Comment (singvarr) [inline src/components/TableCell/TableCell.tsx]: I shouldn't have `key`

File: src/components/TableCell/TableCell.tsx [hunk 29]
```diff
@@ -1,19 +1,22 @@
...
+import * as classes from './styles';
 
 interface TableCellProps {
   weather: ForecastBody;
-  displayDate?: boolean;
+  isDateShown?: boolean;
 }
 
-export default function TableCell({ weather, displayDate }: TableCellProps) {
+export default function TableCell({ weather, isDateShown }: TableCellProps) {
```
Comment (singvarr) [inline src/components/TableCell/TableCell.tsx]: `WeatherForecastTableCell`

File: src/components/Forecast/Forecast.tsx [hunk 30]
```diff
@@ -30,30 +32,39 @@ export default function Forecast() {
   const [selectedLocation, onSelectLocation] = useState<Location | null>(null);
   return (
     <>
+      <div css={classes.wrap}>
```
Comment (singvarr) [inline src/components/Forecast/Forecast.tsx:34]: Use not fragment but `div` as a wrapper for component's markup

File: src/utils/formatDate.ts [hunk 31]
```diff
@@ -1,6 +1,6 @@
 import { format } from 'date-fns';
 
-const dateFormat = 'dd MMM';
+const dateFormat = 'dd MMMM';
```
Comment (singvarr) [inline src/utils/formatDate.ts]: use CONSTANT_CASE

File: src/components/WeatherWidget/styles.ts [hunk 32]
```diff
@@ -0,0 +1,19 @@
...
+export const weatherTable = (theme: any) => css`
+  margin: 15px;
+  ${theme.breakpoints.down('md')} {
+    display: none;
+  }
+`;
+
+export const weatherList = (theme: any) => css`
+  margin: 15px;
+  ${theme.breakpoints.up('bd')} {
```
Comment (singvarr) [inline src/components/WeatherWidget/styles.ts]: fix typo, please

File: src/components/WeatherTable/WeatherTable.tsx [hunk 33]
```diff
@@ -0,0 +1,87 @@
...
+          `${findCountryNameByCode(location.country)}, ${location.name}`}
+      </Typography>
+      <TableContainer
+        component={Paper}
+        css={classes.tableContainer}>
+        <Table aria-label='5-Day Weather Forecast'>
+          <TableHead>
+            <TableRow>
+              <MUITableCell css={classes.tableHeadCell}>
+                Date/Hours
```
Comment (singvarr) [inline src/components/WeatherTable/WeatherTable.tsx]: Just time

File: src/components/WeatherTable/WeatherTable.tsx [hunk 34]
```diff
@@ -0,0 +1,87 @@
...
+
+export default function WeatherTable({ forecast, location, className }: Props) {
+  return (
+    <Box className={className}>
+      <Typography
+        component='p'
+        css={classes.tableTitle}
+        variant='subtitle1'>
+        5-Day Weather Forecast for&nbsp;
+        {location &&
```
Comment (singvarr) [inline src/components/WeatherTable/WeatherTable.tsx]: Redundant check

File: src/components/WeatherList/WeatherList.tsx [hunk 35]
```diff
@@ -0,0 +1,128 @@
...
+  };
+
+  return (
+    <List
+      disablePadding
+      component='ul'
+      className={className}
+      css={classes.headList}
+      subheader={
+        <ListSubheader component='p'>
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx]: Extract it to a separate component and reuse for a table and list

File: src/components/WeatherList/WeatherList.tsx [hunk 36]
```diff
@@ -0,0 +1,128 @@
...
+  const handleClick = (index: number) => {
+    const newOpenItems = [...openItems];
+    newOpenItems[index] = !newOpenItems[index];
+    setOpenItems(newOpenItems);
+  };
+
+  return (
+    <List
+      disablePadding
+      component='ul'
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx]: I think it's `ul` by default

File: src/components/WeatherTable/WeatherTable.tsx (hunk 10 above)
Comment (coding-vibe) [reply src/components/WeatherTable/WeatherTable.tsx]: It indicates the quantity of available weather data for today. We need this information to determine how many cells we should skip.

File: src/components/WeatherList/WeatherList.tsx (hunk 14 above)
Comment (coding-vibe) [reply src/components/WeatherList/WeatherList.tsx]: Yes, because I use it to control which list item should be collapsed or expanded. But maybe there is a better way to do this :)

File: src/components/Select/styles.ts (hunk 22 above)
Comment (coding-vibe) [reply src/components/Select/styles.ts]: I added styles for devices with extension less than 480px

File: src/components/ThemeProvider/ThemeProvider.tsx (hunk 25 above)
Comment (coding-vibe) [reply src/components/ThemeProvider/ThemeProvider.tsx]: Now I removed some overrides of typography and left only useful. 
About breakpoints - I need only 2 breakpoints to override, but it is a requirement of MUI to define all breakpoints - 'If you change the default breakpoints's values, you need to provide them all'.

File: src/components/Forecast/styles.ts [hunk 37]
```diff
@@ -1,13 +1,23 @@
 import { css } from '@emotion/react';
+import { Theme } from '@mui/material';
 
-export const languageSelect = css`
-  margin-top: 10px;
+export const caption = (theme: Theme) => css`
+  margin: 15px 0px 0px 15px;
```
Comment (singvarr) [inline src/components/Forecast/styles.ts]: Your styles shouldn't be overriden on `sm` breakpoint; you shouldn't have overlap of the same properties. Put it to `theme.breakpoints.up('sm')`
Fix this pattern everywhere in your changes

File: src/components/WeatherList/WeatherList.tsx (hunk 14 above)
Comment (singvarr) [reply src/components/WeatherList/WeatherList.tsx]: Yeah, we cannot make it uncontrolled, unfortunately. But we should make it better: index isn't reliable identifier for your items; use timestamps
Also, I recommend using `Set` in order to simplify manipulations with iterable

**Comment by singvarr** [comment]
Screenshot 2023-09-18 at 22 16 34

Check markup when a validation if failed, please

File: src/components/ThemeProvider/ThemeProvider.tsx (hunk 25 above)
Comment (singvarr) [reply src/components/ThemeProvider/ThemeProvider.tsx]: Ok. Add comment about it, please

File: src/constants/theme.ts [hunk 38]
```diff
@@ -0,0 +1,37 @@
...
+      sm: 480,
+      md: 1000,
+      lg: 1200,
+      xl: 1536,
+    },
+  },
+  palette: {
+    secondary: {
+      main: '#49484a',
+      light: '#E0E0E0',
```
Comment (singvarr) [inline src/constants/theme.ts]: Lowercase me

File: src/components/HistoricalWeatherForm/styles.ts [hunk 39]
```diff
@@ -1,23 +1,55 @@
...
+export const startDateField = (theme: Theme) => css`
+  margin-right: 30px;
 
   &::after {
     position: absolute;
     content: '-';
     font-size: 35px;
     color: #707070;
-    left: 225px;
+    left: 105%;
```
Comment (singvarr) [inline src/components/HistoricalWeatherForm/styles.ts]: -5%?

File: src/components/HistoricalWeatherForm/styles.ts [hunk 40]
```diff
@@ -1,23 +1,55 @@
...
-export const startDateField = css`
-  width: 200px;
+export const startDateField = (theme: Theme) => css`
+  margin-right: 30px;
 
   &::after {
     position: absolute;
     content: '-';
     font-size: 35px;
     color: #707070;
```
Comment (singvarr) [inline src/components/HistoricalWeatherForm/styles.ts:19]: Colour should be after positioning styles.
`top` should be before `left`

File: src/components/HistoricalWeatherForm/styles.ts [hunk 41]
```diff
@@ -1,23 +1,55 @@
...
-    left: 225px;
+    left: 105%;
     top: 50%;
     transform: translateY(-50%);
   }
+
+  ${theme.breakpoints.down('sm')} {
+    margin-right: 15px;
+
+    &::after {
```
Comment (singvarr) [inline src/components/HistoricalWeatherForm/styles.ts]: Nest responsive style for pseudoelement in pseudo-element

File: src/components/HistoricalWeatherForm/styles.ts [hunk 42]
```diff
@@ -1,23 +1,55 @@
...
+      left: 103%;
+      font-size: 30px;
+    }
+
+    & .MuiInputBase-input {
+      font-size: 13px;
+    }
+  }
+
+  //I need to change top positioning when error appears, but I have some problems with that, so I left that comment as a reminder
```
Comment (singvarr) [inline src/components/HistoricalWeatherForm/styles.ts]: Don't forget to remove me ;)

File: src/components/HistoricalWeatherForm/styles.ts [hunk 43]
```diff
@@ -1,23 +1,55 @@
...
+import { Theme } from '@mui/material';
 
-export const location = css`
-  width: 460px;
+export const form = (theme: Theme) => css`
+  display: flex;
+  flex-direction: column;
+  width: 435px;
+  gap: 15px;
+  ${theme.breakpoints.down('sm')} {
```
Comment (singvarr) [inline src/components/HistoricalWeatherForm/styles.ts]: Add a blank line before responsive opening line everywhere

File: src/components/HistoricalWeatherTable/HistoricalWeatherTable.tsx [hunk 44]
```diff
@@ -0,0 +1,77 @@
...
+                <TableRow key={index}>
+                  {index === 0 &&
+                    Array.from({ length: emptyCellsCount }).map((_, idx) => (
+                      // We should leave some cells empty because user chooses historical forecast for specific dates and some days of week should be skipped
+                      <MUITableCell key={idx} />
+                    ))}
+                  {weeklyWeather.map((dailyWeather, idx) => (
+                    <WeatherTableCell
+                      isDateShown
+                      key={idx}
```
Comment (singvarr) [inline src/components/HistoricalWeatherTable/HistoricalWeatherTable.tsx]: Don't use index here

File: src/components/WeatherList/WeatherList.tsx [hunk 45]
```diff
@@ -0,0 +1,125 @@
...
+          css={classes.headListTitle}
+          name={location.name}
+          text='5-Day'
+        />
+      }>
+      {forecast?.map(([date, weather], index) => (
+        <List
+          disablePadding
+          key={date}>
+          <ListItem
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx:55]: Add a top border for list item on the forecast page

File: src/components/WeatherList/WeatherList.tsx [hunk 46]
```diff
@@ -0,0 +1,125 @@
...
+                weather: [{ description, icon }],
+              } = hourlyWeather;
+              const hour = format(dt * 1000, 'hh:00 a');
+              return (
+                <List
+                  disablePadding
+                  key={idx}>
+                  <ListItem disablePadding>
+                    <ListItemText css={classes.listItemText}>
+                      <p>{hour}</p>
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx]: `span` is semantically better

File: src/components/WeatherSearchCaption/styles.ts [hunk 47]
```diff
@@ -0,0 +1,22 @@
...
+
+export const text = (theme: Theme) => css`
+  color: ${theme.palette.secondary.main};
+  ${theme.breakpoints.down('sm')} {
+    font-size: 9px;
+  }
+`;
+
+export const icon = (theme: Theme) => css`
+  fill: ${theme.palette.secondary.main};
```
Comment (singvarr) [inline src/components/WeatherSearchCaption/styles.ts]: use color

File: src/components/WeatherTable/WeatherTable.tsx [hunk 48]
```diff
@@ -0,0 +1,79 @@
...
+                  align='center'
+                  css={classes.tableHeadCell}
+                  key={hour}>
+                  {hour}
+                </MUITableCell>
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {forecast?.map(([date, weather], index) => {
```
Comment (singvarr) [inline src/components/WeatherTable/WeatherTable.tsx]: You don't need optional chaning

File: src/components/WeatherList/styles.ts [hunk 49]
```diff
@@ -0,0 +1,58 @@
...
+  & .MuiListItemText-primary {
+    color: ${theme.palette.secondary.main};
+    font-weight: 500;
+    ${theme.breakpoints.down('sm')} {
+      font-size: 10px;
+    }
+  }
+`;
+
+export const listItemText = (theme: Theme) => css`
```
Comment (singvarr) [inline src/components/WeatherList/styles.ts]: These styles can be reused and `mainListItemText` too

**Review by singvarr — COMMENTED** [review]
We are almost ready

**Review by singvarr — COMMENTED** [review]
We are almost ready. I like these changes

File: src/components/HistoricalWeatherForm/styles.ts [hunk 50]
```diff
@@ -1,23 +1,62 @@
...
+      left: 103%;
+      font-size: 30px;
+    }
   }
 `;
 
-export const endDateField = css`
-  width: 200px;
+export const endDateField = (theme: Theme) => css`
+  ${theme.breakpoints.down('sm')} {
```
Comment (singvarr) [inline src/components/HistoricalWeatherForm/styles.ts:32]: Should be reused

File: src/components/Select/styles.ts [hunk 51]
```diff
@@ -1,5 +1,18 @@
 import { css } from '@emotion/react';
+import { Theme } from '@mui/material';
 
-export const select = css`
-  min-width: 200px;
+export const select = (theme: Theme) => css`
+  ${theme.breakpoints.up('sm')} {
+    width: 435px;
```
Comment (singvarr) [inline src/components/Select/styles.ts]: width limitation to div; these widths should be in theme

File: src/components/HistoricalWeatherList/HistoricalWeatherList.tsx [hunk 52]
```diff
@@ -0,0 +1,94 @@
...
+      }>
+      {weeklyForecast?.map((weeklyWeather) =>
+        weeklyWeather.map((dailyWeather) => {
+          const {
+            dt,
+            main: { humidity, temp },
+            weather: [{ description, icon }],
+          } = dailyWeather;
+          return (
+            <ListItem disablePadding>
```
Comment (singvarr) [inline src/components/HistoricalWeatherList/HistoricalWeatherList.tsx]: add `key`

File: src/components/HistoricalWeatherList/HistoricalWeatherList.tsx [hunk 53]
```diff
@@ -0,0 +1,53 @@
...
+      css={classes.list}
+      subheader={
+        <WeatherContentHeader
+          country={country}
+          css={classes.listTitle}
+          name={name}
+          text='Historical'
+        />
+      }>
+      {weeklyForecast?.map((weeklyWeather) =>
```
Comment (singvarr) [inline src/components/HistoricalWeatherList/HistoricalWeatherList.tsx]: You don't need optional chaining. Check other places

File: src/components/WeatherList/WeatherList.tsx [hunk 54]
```diff
@@ -0,0 +1,88 @@
...
+import * as classes from './styles';
+
+interface Props {
+  forecast: Forecast;
+  location: Location;
+  className?: string;
+}
+
+export default function WeatherList({ forecast, location, className }: Props) {
+  // const { selectedTemperatureUnit } =
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx]: Don't forget to remove it

File: src/components/WeatherList/WeatherList.tsx [hunk 55]
```diff
@@ -0,0 +1,88 @@
...
+  //   useContext<SettingsContextType>(SettingsContext);
+  const forecastDates = new Set<string>();
+  forecast.forEach(([date]) => forecastDates.add(date));
+  const [openItems, setOpenItems] = useState<Set<string>>(forecastDates);
```
Comment (singvarr) [inline src/components/WeatherList/WeatherList.tsx]: Make me simpler: `new Set(...forecast)`

**Review by singvarr — APPROVED** [review]
LGTM. Fix these 3 minor comments and merge the branch
